fees/euler/index.ts

```diff
@@ -0,0 +1,123 @@
+}
+
+const getVaults = async ({createBalances, api, fromApi, toApi, getLogs, chain}: FetchOptions, {
+    dailyFees,
+    dailyRevenue,
+}: {
+    dailyFees?: sdk.Balances,
+    dailyRevenue?: sdk.Balances,
+}) => {
+
+    if (!dailyFees) dailyFees = createBalances()
+    if (!dailyRevenue) dailyRevenue = createBalances()
+    const vaultLength = await fromApi.call({target: eVaultFactories[chain], abi: eulerFactoryABI.vaultLength})
+    if(vaultLength === 0) return {}
+    const vaults = await fromApi.call({target: eVaultFactories[chain], abi: eulerFactoryABI.getProxyListSlice, params: [0, vaultLength]})
+    const underlyings = await fromApi.multiCall({calls: vaults, abi: eulerVaultABI.asset})
+    underlyings.forEach((underlying, index) => {
+        if (!underlying) underlyings[index] = '0x0000000000000000000000000000000000000000'
+    })    
+
+    const accumulatedFeesStart = await fromApi.multiCall({calls: vaults, abi: eulerVaultABI.accumulatedFees})
+    const accumulatedFeesEnd = await toApi.multiCall({calls: vaults, abi: eulerVaultABI.accumulatedFees})
+    const interestAccumulatorStart = await fromApi.multiCall({calls: vaults, abi: eulerVaultABI.interestAccumulator})
+    const interestAccumulatorEnd = await toApi.multiCall({calls: vaults, abi: eulerVaultABI.interestAccumulator})
+    const totalBorrows = await toApi.multiCall({calls: vaults, abi: eulerVaultABI.totalBorrows})
+    const dailyInterest = totalBorrows.map((borrow, i) => borrow * (interestAccumulatorEnd[i] - interestAccumulatorStart[i]) / interestAccumulatorStart[i])
+
+    const logs = (await getLogs({targets: vaults, eventAbi: eulerVaultABI.convertFees, flatten: false})).map((vaultLogs) => {
+        if (!vaultLogs.length) return 0n;
+        let totalShares = 0n;
+        for (const log of vaultLogs) {
+            totalShares += (log.protocolShares + log.governorShares);
+        }
+        return totalShares;
+    });
+ 
+    //calculate (accumulatedFeesEnd - accumulatedFeesStart) + totalShares from convertFees
+    const accumulatedFees = accumulatedFeesEnd.map((fees, i) => {
+        const feesEnd = BigInt(fees.toString());
+        const feesStart = BigInt(accumulatedFeesStart[i].toString());
+        return feesEnd - feesStart + logs[i];
+    });
+
+    //we then convert the accumulatedFees to asset by calling convertToAssets at the end therefore we won't have any problem with conversion rate changing
+    const totalAssets = await toApi.multiCall({
+        calls: accumulatedFees.map((fees, i) => ({
+            target: vaults[i],
+            params: [fees.toString()]
+        })),
+        abi: eulerVaultABI.convertToAssets,
+    });
+
+    totalAssets.forEach((assets, i) => {
+        dailyFees.add(underlyings[i], dailyInterest[i])
+        dailyRevenue.add(underlyings[i], assets)
+    })
+    const dailySupplySideRevenue = dailyFees.clone()
```

Comment:
nit: shall this variable be called `dailySupplySideRevenue`? to be consistent I guess it should be called something like `dailySupplySideFees`?

Reply:
you are right, but it is a bit late to fix now, though we could change this display label for this field in the website for all protocols